### PR TITLE
The End of the Galactic Materials Market: Economy shakeup

### PR DESCRIPTION
**Pull request by Ephemeralis**
## About The Pull Request

tldr;

- The GMM crate has been removed from cargo, functionally removing it from the game except for when admins want it there.
- The amount of credits a character joins the round with has doubled from 5 paychecks to 10. You can expect to have 50 minutes of paychecks available to you from the moment you enter the round instead of the usual 25.
- Using departmental vendors grants you a 50% discount instead of a 20%.

We need to be honest with ourselves: the in-game economy sucks.

There's several highly publicized ways to produce an absurd quantity of credits that involve minimal interaction with other players, this is not news to anyone who has played for longer than a few days. One of these methods in particular has proven *absolutely disastrous* for an entire department: the Galactic Materials Market.

The GMM has damaged cargo gameplay more than any other addition in the past five years. With our three hour round outlook, anyone with even a basic understanding of arbitrage (buy low, sell high) and the willingness to forego **all** interaction with the round to stare at the GMM terminal to watch for the 60 second market ticks can produce millions of credits entirely without risk. Quartermasters have pivoted from dodgy logistics officers to silent, statuesque financiers interested only in numbers on a terminal. I have seen vault break-ins ignored because quartermasters are able to produce money faster than someone *draining the bank terminal* can steal it.

No more. This PR removes the GMM terminal from the cargo market, rendering its inclusion in rounds at the sole discretion of the administration. Cargo is once again free to harp at each other over crates, export actual, physical items, and generally grub the station for things to pawn for a profit, as the Space Gods intended.

With that out of the way, we arrive at the next contention point: bounties.

I have made a concerted push over the past few months to try and ensure that the basic chargen loadout options provide a wide gamut of outfits, equipment and tools to tide people over without them feeling compelled to waste time printing things from a departmental autolathe for the Funny Golden Cubes.

Basically nobody enjoys doing bounties. The boffins in devcont have some ideas(tm) about what we can do about bounties in the long term, but this'll require some elbow grease.

To alleviate this, the starting number of paychecks has been doubled from 5 to 10. In practice, this shakes out to roughly two bounties worth of money (for anyone but medical/supply, who have absolutely cracked bounties). You can buy more gear from the get-go and spend more time plotting things to do in round, or interacting with people.

Additionally, departmental vendors now have a whopping 50% discount for people who work in the department, up from 20%. This makes gearing up cheaper and also opens up opportunity for characters to flip gear off to other characters at a slight profit.

## How This Contributes To The Nova Sector Roleplay Experience

The small essay above covers most of the scope, but the summary of it is: with the GMM gone, cargo is free to do fun and dodgy cargo things to earn obscene quantities of money (such as selling bricks of cocaine or contraband gear).

Individual players will have less of a need to do bounties to purchase the lingering non-loadout necessities for their characters, and the departmental vendor discount makes gearing up in new roles cheaper, while also allowing opportunities for player-to-player arbitrage in terms of selling off gear.

## Proof of Testing

Basic number tweaking - CI will let us know if not.

## Changelog

:cl: yooriss
del: The Galactic Materials Market can no longer be purchased from cargo. Admins are able to spawn it in, but it will not generally be accessible in normal rounds anymore.
balance: Characters now start with 10 paychecks in their bank account instead of 5, functionally doubling the credits people enter the round with.
balance: Departmental vendors now offer a 50% discount to employees of their given department, up from 20%.
/:cl:

### CODE REVIEW

**Comment by Nerev4r** [comment]
As a long-term Cargo player since basically forever, there's two kinds of rounds I really don't like.
Rounds where everyone stares at the GMM screen like it's the fucking hypnotoad and we're not actually doing anything character-related, or 
image
rounds like this.
The GMM isn't solely responsible for these things, but it's a top contributor for sure.

**Comment by Ephemeralis** [comment]
> not a fan of this because while yes it can make a lot of money, by far the biggest use is fixing material shortages, lately they have been extremely bad to the point of having little to no of a certain material(s) 2 hours into the round, which the GMM can fix, there's been rounds where the GMM was the only reason why the station gets mats in the first place, unless you're removing all of arcmining so its back to the old version, no.
> 
> also this means people are just gonna sell mats on the cargo shuttle to make money that way

Will probably be bringing the GMM back with some revisions to this later today but with severely capped ceilings so that arbitrage is nowhere near as prominent.

Also, materials selling on the shuttle is supposed to be governed by elasticity iirc so people can't produce hail mary quantities of credits with it.

This all being said, some discussion in devcont raised that cargo doesn't have a lot of salient endgame outside of amassing huge sums of credits to buy stupid things with, so maybe giving them gameplay stuff to fill the gap is warranted as well.

**Comment by FlufflesTheDog** [comment]
>Departmental vendors now offer a 50% discount to employees of their given department, up from 20%.

The current discount is 80%. This makes vendors in your department more expensive.

I'll also second the opinion that cargo should have a way to buy materials- iron and glass, at the very least, since they're needed for literally everything and in such large volumes.

**Comment by Ephemeralis** [comment]
> > Departmental vendors now offer a 50% discount to employees of their given department, up from 20%.
> 
> The current discount is 80%. This makes vendors in your department more expensive.

Whoops. Good catch. Reverting that.

**Comment by OrbisAnima** [comment]
Recommend you re-add the removed material crates, that is, 50 glass, 50 iron at cargo crate cost x2 (400 credits) each, the 20 plasteel crate that I cant remember how much it went for, which will help significantly not only shortages but also assistant construction, plus its how it was before the GMM arrived.

Love the PR!

**Comment by OrbisAnima** [comment]
A question too since it's been a while from when I reviewed the payment code, would doubling the round starting money affect the mid round joining money?

**Comment by Zergspower** [comment]
> not a fan of this because while yes it can make a lot of money, by far the biggest use is fixing material shortages, lately they have been extremely bad to the point of having little to no of a certain material(s) 2 hours into the round, which the GMM can fix, there's been rounds where the GMM was the only reason why the station gets mats in the first place, unless you're removing all of arcmining so its back to the old version, no.
> 
> also this means people are just gonna sell mats on the cargo shuttle to make money that way

Arcmining has given us tenfold more materials than what we had before, so long as miners do their jobs. I also purposefully made it so you can still mine normally so the GMM is not needed here for materials.

**Comment by chemistrymain3** [comment]
Truthfully, simply adding materials crates to cargo so that they can *buy* materials is also a valid option.

Why not allow cargo to import ten sheets of diamond in a pinch? Or glass, or plastic, or iron. Just make it expensive.

That way, even if all the miners are dead, everyone doesn't suffer cock & ball torture.

**Comment by OrbisAnima** [comment]
We got the bluespace miner for 10k that can be set anywhere, then the ore stumper that can be set on the planet at the cost of 50kw and 1-2k and produces everything but strange rocks I think, and lastly we have the ant farms that cannot produce diamonds and bluespace crystals. We also got xenobiology, bitrunning and space tiding as sources of minerals.

Not to mention the automatic vent that we start with that prodives glass and iron.

**Comment by Ephemeralis** [comment]
> A question too since it's been a while from when I reviewed the payment code, would doubling the round starting money affect the mid round joining money?

Yes, this rolls at shift-start or game join. Everyone's baseline increases.

**Comment by annikaRU** [comment]
I feel this is more trouble than it's worth, as someone who does cargo for their primary role, the market isn't used as often as it used to be, there are a few individuals that know how to use it well and can easily make millions but thats still late shift 2 hours in. Hour mark you would have around 200,000 which is perfectly acceptable. Most problems can just be solved by setting policy instead of removing something as was done prior.

and the instances where it is abused are very niche cases and dont happen that often anymore,  infact Ive seen cargo have 500,000 around 8 times and over a million twice which when they had a million was used for the sole purpose of purchasing a bike.

Market could use changes yes, should be removed hard no unless mining is made easier and you increase the value of items that cargo is able to sell because the reason it's used is its the easiest way to make large amounts of money for potential projects, budget increases and other useless stuff.

**Comment by Hatterhat** [comment]
> Hour mark you would have around 200,000 which is perfectly acceptable.

that's the equivalent of one thousand standard crates or four thousand crew paychecks (the rough basic units of the tg economy) i think you are grossly underestimating how far that goes

**Comment by davidchan13** [comment]
Generally agree with this PR, don't entirely agree with removing it outright but Cargos top money makers should always require other departments to aid them imo. Outside the scope of this pr but Id like bounties to be looked at in the near future to give more incentive for other departments to work with cargo to generate income, it should feel like win win with cargo getting money and the other departments getting something for their time.

For now though, removing or disabling the GMM is a net positive. Bringing it back someday might be possible with nerfs and require more active work and resources to keep it running and up to date, but I point to my above stance that cargos best abilities to make money should always be inter department affairs, and not something cargo can do by just order machines other departments use (like how they always seem get botany setups to feed their own coloy core and biogen machines)

**Comment by CliffracerX** [comment]
I'd propose disabling the ability to sell mats through the GMM, while keeping the option to buy them (and the opportunity to try and game the market by getting mats at wildly cheap prices) - removes it as a ginormous draw to spam moneymaking, whilst keeping its most useful function as a stopgap for keeping the station's materials supply steady when mining/bitrunning is on low population.  Just killing it outright and switching back to the limited & overpriced materials crate isn't fantastic, especially for lowpop shifts where people aren't gonna be around in the roles to make mats.

**Review by SomeRandomOwl — CHANGES_REQUESTED** [review]
I like this PR, i think its needed, but it'd be easier to disable the ability to sell materials, and keep the ability to buy. IIRC, youd have to change the GMM's attack_by proc and disable the ability for it to create stock cubes. 

That way you can still buy from it just not sell. 

If you are insistent on disabling it entirely, then the supply packs of materials can be added modularly and not non-modularly. You can stick it in one of our modular files which adds supply packs or just make a new one. That is a change i will require if you insist on fully disabling the GMM. Otherwise I'd accept simply disabling its ability to sell/create stock cubes, preferably with a feedback message saying that the market doesn't accept sales or something.

**Comment by Ephemeralis** [comment]
I can retain the ability to buy and sell but *hugely* clamp the price variant range so that any divergent shifts are like, in the order of 5-10% instead of 50-250%. That functionally removes the ability to perform meaningful arbitrage with the GMM, but lets the crew potentially offload large quantities of excess minerals for cargo budget if they want to.

Fair compromise?

**Comment by Iamgoofball** [comment]
>The GMM has damaged cargo gameplay more than any other addition in the past five years.

This is patently false. Stock /tg/ cargo bays aren't regularly capable of affording bicycles even with the GMM. You guys busted up cargo on your own, this is the result of years of minor changes nickel and diming the stability of your Cargo economy.

**Comment by Iamgoofball** [comment]
I mean, you can just tweak the numbers to accommodate for the fact your rounds are 3x as long.

**Comment by CliffracerX** [comment]
> I mean, you can just tweak the numbers to accommodate for the fact your rounds are 3x as long. This really isn't a hard problem to solve.

This is a band-aid solution and we've already tried a lot of those.  GMM being free money through basic min-maxing is just a little busted & even more uninteresting than bounties, and turning the numbers down result in it *still* being busted and uninteresting, just even slower and more grindy than it already is.  Slowing down the grind to cargo's nonexistent late-game doesn't result in a more interesting late-game for the station or more interesting mechanics for the crew working to get there, it just makes it grindy.

Until such time as cargo economy actually matters in a positive way rather than just being a source of strife as it arms up to independence off GMM profits every round, just nuking GMM as a money source in favour of being an alternative route to feed the insatiable hunger of a hundred people with protolathes is probably the best route.

**Comment by SomeRandomOwl** [comment]
Requested by a headmin to have golden personally review this.

**Comment by Ephemeralis** [comment]
@GoldenAlpharex any path forward with this? willing to make any changes needed

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
So, for starters, I think `devcont` has to be the funniest way I've seen someone shorten up development-contributor, heh.

Secondly, and more importantly, I think the Galactic Materials Market could instead be limited to only buy materials from. That way, we retain the ability to get materials that could be in shorter supply on station, with still some randomness to their price, but there's no way to abuse it because you can't sell it back for a profit.

I think that would fix the main problem that I have with this PR, which is that now several materials are once again not purchasable anymore, such as diamonds, bluespace crystals, titanium, gold and silver.

File: code/controllers/subsystem/stock_market.dm [hunk 1]
```diff
@@ -1,9 +1,9 @@
-#define MINIMUM_VALUE_MULT 0.95 // used in price_minimum declarations, /tg/ default is 0.5
-#define MAXIMUM_VALUE_MULT 1.15 // used in prixe_maximum declarations, /tg/ default is 3
+#define MINIMUM_VALUE_MULT 0.5 // NOVA EDIT: used in price_minimum declarations, /tg/ default is 0.5
```
Comment (GoldenAlpharex) [inline code/controllers/subsystem/stock_market.dm:1]: If you're keeping the values at the /tg/ values, then you should probably make this change happen upstream instead. There's no need to expose defines downstream if we're not going to be changing them. I also don't think they'd refuse this change if someone made a PR for it, it's just a small code improvement.

File: modular_nova/modules/economy/code/materials_market.dm [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Behavior overrides for the Galactic Materials Market
+
+/obj/machinery/materials_market
+	desc = "This machine allows the user to buy sheets of minerals \
+	across the system. It once allowed the sale of excess minerals as well, but after the Great Crash of 2564, nobody buys materials from the frontier anymore. \
+		Prices are known to fluctuate quite often, sometimes even within the same minute. All transactions are final."
+
+/obj/machinery/materials_market/attackby(obj/item/O, mob/user, params)
+	if(is_type_in_list(O, exportable_material_items))
+		say("Selling materials to the GMM is no longer available due to volatile logistics conditions in frontier space. Please export your materials via standard Union-NT cargo arrangements.")
```
Comment (GoldenAlpharex) [inline modular_nova/modules/economy/code/materials_market.dm:10]: No single-letter variables.
```suggestion
/obj/machinery/materials_market/attackby(obj/item/attacking_item, mob/user, params)
	if(is_type_in_list(attacking_item, exportable_material_items))
		say("Selling materials to the GMM is no longer available due to volatile logistics conditions in frontier space. Please export your materials via standard Union-NT cargo arrangements.")
```

File: modular_nova/modules/economy/code/materials_packs.dm [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Readds the materials crates the GMM removed, but mostly for things like assistant projects.
+
+/datum/supply_pack/materials/glass50
```
Comment (GoldenAlpharex) [inline modular_nova/modules/economy/code/materials_packs.dm:3]:
```suggestion
/datum/supply_pack/materials/glass_fifty
```

File: modular_nova/modules/economy/code/materials_packs.dm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	name = "50 Glass Sheets"
+	desc = "Let some nice light in with fifty glass sheets!"
+	cost = CARGO_CRATE_VALUE * 2
+	contains = list(/obj/item/stack/sheet/glass/fifty)
+	crate_name = "glass sheets crate"
+
+/datum/supply_pack/materials/iron50
```
Comment (GoldenAlpharex) [inline modular_nova/modules/economy/code/materials_packs.dm:10]:
```suggestion
/datum/supply_pack/materials/iron_fifty
```

File: modular_nova/modules/economy/code/materials_packs.dm [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	name = "50 Iron Sheets"
+	desc = "Any construction project begins with a good stack of fifty iron sheets!"
+	cost = CARGO_CRATE_VALUE * 2
+	contains = list(/obj/item/stack/sheet/iron/fifty)
+	crate_name = "iron sheets crate"
+
+/datum/supply_pack/materials/plasteel20
```
Comment (GoldenAlpharex) [inline modular_nova/modules/economy/code/materials_packs.dm:17]:
```suggestion
/datum/supply_pack/materials/plasteel_twenty
```

File: modular_nova/modules/economy/code/materials_packs.dm [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+	name = "20 Plasteel Sheets"
+	desc = "Reinforce the station's integrity with twenty plasteel sheets!"
+	cost = CARGO_CRATE_VALUE * 15
+	contains = list(/obj/item/stack/sheet/plasteel/twenty)
+	crate_name = "plasteel sheets crate"
+
+/datum/supply_pack/materials/plasteel50
```
Comment (GoldenAlpharex) [inline modular_nova/modules/economy/code/materials_packs.dm:24]:
```suggestion
/datum/supply_pack/materials/plasteel_fifty
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/stock_market.dm (hunk 1 above)
Comment (Ephemeralis) [reply code/controllers/subsystem/stock_market.dm:1]: I think we should leave these in for the moment, principally during the testmerge phase since I suspect we're going to need to tweak the maximum value multiplier quite significantly, if we do need to tweak anything at all.

Comment (GoldenAlpharex) [reply code/controllers/subsystem/stock_market.dm:1]: I'm willing to allow that, if you also bring it upstream so that can be there instead, and we don't have to maintain some defines like that in the future.

Comment (Ephemeralis) [reply code/controllers/subsystem/stock_market.dm:1]: That should be no problem, yeah.
